attributes/defaults.rb

```diff
@@ -54,6 +66,26 @@
   default['mysqld']['my.cnf']['mysqld_safe']['skip_log_error'] = true
 end
 
+# Password columns for user passwords
+case node['mysqld']['db_install']
+when 'mariadb'
+  default['mysqld']['pwd_col'] = 'Password'
+when 'percona'
+  if node['mysqld']['repository']['percona']['version'] == '5.6'
+    default['mysqld']['pwd_col'] = 'Password'
+  else
+    default['mysqld']['pwd_col'] = 'authentication_string'
+  end
+when 'mysql'
+  if node['platform'] == 'debian' && node['platform_version'].to_f < 9.0
```

Comment:
Sounds better, this was only to use the right column when setting the password and using default distro packages, in this case, debian jessie comes with mysql 5.5, and stretch at the moment with 5.6 which i believe will end having 5.7 which support authentication_string.

Using chef_sugar maybe does it make it more readable, but adds a dependency.